src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/_utils.py

```diff
@@ -242,6 +242,49 @@ def _parameters(registry_name,
     return parameters
 
 
+def get_credentials(cmd, auth_mode=None, credentials=None):
+    """Get the credential object from the input
+    :param str auth_mode: The login mode for the source registry
+    :param [] credentials: The list of credentials
+    """
+    # pylint: disable=line-too-long
+    Credentials, SourceRegistryCredentials, CustomRegistryCredentials, SourceRegistryLoginMode, SecretObject, SecretObjectType = cmd.get_models('Credentials', 'SourceRegistryCredentials', 'CustomRegistryCredentials', 'SourceRegistryLoginMode', 'SecretObject', 'SecretObjectType')
+
+    if auth_mode is None:
+        auth_mode = SourceRegistryLoginMode.default
+
+    sourceRegistryCredentials = SourceRegistryCredentials(login_mode=auth_mode)
+
+    customRegistries = {}
+    if credentials is None:
+        credentials = []
+
+    for credential in credentials:
+        cred_split = credential.split(';', 2)
+        if len(cred_split) != 3:
+            raise CLIError(
+                "Please provide the credentials in the form of "
+                "'registryName;username;secret'. Incorrect credentials: '{}'".format(credential)
+            )
+
+        # { 'loginServer': { username: <SecretObject>, password: <SecretObject> } }
+        customRegistries[cred_split[0]] = CustomRegistryCredentials(
```

Comment:
better to trim the extra space after split. for example, if user type something like credential='reg1; username ;  password '

Reply:
>             better to trim the extra space after split. for example, if user type something like credential='reg1; username ;  password '

Yes, You may have to split using removing empty entries